git-annex-remote-rclone

```diff
@@ -258,18 +285,102 @@ while read -r line; do
 				;;
 			esac
 		;;
+		TRANSFEREXPORT)
+			if [ -z "$GOT_EXPORT" ]; then
+				abort_annex_race_bug "received TRANSFEREXPORT without prior EXPORT" || exit 2
+			fi
+			GOT_EXPORT=""
+
+			op="$2"
+			key="$3"
+			file="$4"
+			case "$op" in
+				STORE)
+					# Store the file to a temporary location next to the final destination,
+					# and then move it to the final destination, hopefully atomically.
+					# 16 bytes from /dev/urandom is 128 bits, or 32 hexdigits after hashing
+					rand=$(head -c 16 /dev/urandom |git hash-object --stdin)
+					tmploc="${EXPORT_LOC}.tmp-${rand:0:32}"
+
+					# XXX when at all possible, send PROGRESS
+					if [ ! -e "$file" ]; then
+						echo TRANSFER-FAILURE STORE "$key" "asked to store non-existent file $file"
+					elif ! runcmd rclone copyto "$file" "$tmploc"; then
+						echo TRANSFER-FAILURE STORE "$key" "upload failed"
+					else
+						if ! runcmd rclone moveto "$tmploc" "$EXPORT_LOC"; then
+							runcmd rclone delete "$tmploc"
+							echo TRANSFER-FAILURE STORE "$key" "move to final destination failed"
```

Comment:
BTW -- shouldn't we in any case of failure try to remove that `$tmploc` to clean up after ourselves?

Reply:
@Wolf480pl ping on this

Reply:
sorry to nudge, but just today while initiating a new remote for a backup of singularity hub I thought -- wouldn't it be nice if we could export to dropbox... which reminded me about this PR ;)  I really hope you would find time to bring this PR over the finish line, especially since IIRC it is so close.